### DISCUSSION TITLE
关于HLSL的一些疑问

### DISCUSSION THREAD

**Discussion by MikeWang000000** [Blinue/Magpie#690] [Q&A] [answered]
首先感谢Magpie开发者的辛勤劳动！关于HLSL着色器，我有一个疑问。
Magpie将很多GLSL着色器，如Anime4k系列，进行了HLSL移植。
这些移植的HLSL文件是否能应用于MPC-HC或者MPC-BE？这样播放器就能直接进行放大，达到和Magpie一样的效果。

**Comment by Blinue** [comment, marked as answer]
Magpie 中着色器的语法是自创的，不能在别的地方使用，同样 MPC-HC/BE 也有自己的语法。不同软件的着色器是不可能通用的，因为在运行时需要的信息不同。比如 Magpie 的着色器支持传入参数、绘制光标等，播放器却不需要它们。

核心的计算代码确实可以通用，如果想为 MPC-HC/BE 创建着色器，从 Magpie 移植更方便。

**Reply by MikeWang000000** [reply]
非常感谢！
